shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/presentation/modules/login/LoginViewModel.kt

```diff
@@ -0,0 +1,44 @@
+package co.nimblehq.blisskmmic.presentation.modules.login
+
+import co.nimblehq.blisskmmic.data.network.helpers.jsonApiException
+import co.nimblehq.blisskmmic.domain.model.Token
+import co.nimblehq.blisskmmic.domain.usecase.LogInUseCase
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+data class LoginViewState(
+    val isSuccess: Boolean = false,
+    val isLoading: Boolean = false,
+    val error: String? = null
+) {
+    @Suppress("EmptySecondaryConstructor")
+    constructor() : this(false, false, null) {}
+
+    @Suppress("EmptySecondaryConstructor")
+    constructor(error: String?) : this(false, false, error) {}
+}
+
+class LoginViewModel(private val logInUseCase: LogInUseCase) {
```

Comment:
Actually, I think you did too much work in this backend PR. Normally with a backend PR, we just do up to the UseCase level, we leave the VM logic for the integration task. We can execute to call UseCase in local development for testing and POW without polishing and committing its logic yet in this task 🤓

Reply:
Will keep that in mind for the next backend tasks. 
For this one, I included the ViewModel because I would like to show the bare minimum ViewModel structure that will be integrated with iOS.